### PR TITLE
Allow routes to check if user is student or instructor

### PR DESCRIPTION
**Pull request by kmcnellis**
works towards #400

### CODE REVIEW

File: client/components/auth/router.decorator.js [hunk 1]
```diff
@@ -12,20 +12,50 @@ export function routerDecorator($rootScope, $location, Auth) {
...
 
     if (typeof next.authenticate === 'string') {
-      Auth.hasRole(next.authenticate)
-        .then(has => {
-          if (has) {
-            return;
-          }
+      if (next.authenticate === 'student'){
+        return Auth.isStudent()
+          .then(has => {
```
Comment (seveibar) [inline client/components/auth/router.decorator.js]: ??? has?

File: client/components/auth/router.decorator.js [hunk 2]
```diff
@@ -12,20 +12,50 @@ export function routerDecorator($rootScope, $location, Auth) {
...
-          event.preventDefault();
-          return Auth.isLoggedIn()
-            .then(is => {
-              $location.path(is ? '/' : '/login');
-            });
-        });
+            event.preventDefault();
+            return Auth.isLoggedIn()
+              .then(is => {
+                $location.path(is ? '/' : '/login');
```
Comment (seveibar) [inline client/components/auth/router.decorator.js]: is? these variable names are not so great...

**Review by seveibar — CHANGES_REQUESTED** [review]
:penguin: but honestly i think "has" and "is" are pretty low on the good variable name scale

File: client/components/auth/router.decorator.js (hunk 2 above)
Comment (kmcnellis) [reply client/components/auth/router.decorator.js]: Those were the names already used, I just duplicated them for consistency. Do you have a better name in mind?

Comment (seveibar) [reply client/components/auth/router.decorator.js]: haha yea, `isLoggedIn` and `hasRole`, assuming thats what they actually mean (hasRole could be isStudent also?)

**Comment by kmcnellis** [comment]
Fixed

**Comment by kmcnellis** [comment]
@seveibar

**Review by seveibar — APPROVED** [review]

**Pull request merged**